network/p2p/handler.go

```diff
@@ -58,6 +64,28 @@ func (NoOpHandler) CrossChainAppRequest(context.Context, ids.ID, time.Time, []by
 	return nil, nil
 }
 
+// ValidatorHandler drops messages from non-validators
+type ValidatorHandler struct {
+	Handler
+	ValidatorSet ValidatorSet
+}
+
+func (v ValidatorHandler) AppGossip(ctx context.Context, nodeID ids.NodeID, gossipBytes []byte) error {
+	if !v.ValidatorSet.Has(ctx, nodeID) {
+		return ErrNotValidator
```

Comment:
Would it be reasonable to return the height in the error as well?

Reply:
We'd probably have to input the height as an argument into `Has` but I think that looks weird since we don't want this to handle arbitrary height lookup. I think we still have some visibility since this uses the P-Chain so someone could troubleshoot by getting the height off the API.